### PR TITLE
python3: 3.9 -> 3.10

### PR DESCRIPTION
**Pull request by mweinelt**
###### Description of changes

Updating the default version for the next release cycle to Python 3.10.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#cpython-update-schedule

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: doc/languages-frameworks/python.section.md [hunk 1]
```diff
@@ -102,16 +102,19 @@ will still get 1 wrapped Python interpreter. We can start the interpreter
...
+  /nix/store/5gaiacnzi096b6prc6aa1pwrhncmhc8b-python3.9-toolz-0.11.2
+copying path '/nix/store/5gaiacnzi096b6prc6aa1pwrhncmhc8b-python3.9-toolz-0.11.2' from 'https://cache.nixos.org'...
+building '/nix/store/mpn7k6bkjl41fm51342rafaqfsl10qs4-python3-3.9.12-env.drv'...
+created 279 symlinks in user environment
+Python 3.9.12 (main, Mar 23 2022, 21:36:19)
+[GCC 11.3.0] on linux
 Type "help", "copyright", "credits" or "license" for more information.
 >>> import requests
->>>
+>>>
```
Comment (dotlambda) [inline doc/languages-frameworks/python.section.md:117]:
```suggestion
>>>
```

File: pkgs/development/python-modules/sanic/default.nix [hunk 2]
```diff
@@ -24,32 +24,23 @@
...
     rev = "v${version}";
-    sha256 = "0jyl23q7b7fyqzan97qflkqcvmfpzbxbzv0qgygxasrzh80zx67g";
+    sha256 = "sha256-4zdPp3X22dfZ5YlW3G5/OqeUxrt+NiFO9dk2XjEKXEg=";
   };
 
   postPatch = ''
     # Loosen dependency requirements.
     substituteInPlace setup.py \
       --replace "pytest==6.2.5" "pytest" \
       --replace "gunicorn==20.0.4" "gunicorn" \
```
Comment (dotlambda) [inline pkgs/development/python-modules/sanic/default.nix:43]:
```suggestion
      --replace "gunicorn==20.0.4" "gunicorn"
```

File: pkgs/development/python-modules/sanic/default.nix [hunk 3]
```diff
@@ -67,7 +58,6 @@ buildPythonPackage rec {
     gunicorn
     pytest-asyncio
     pytest-benchmark
```
Comment (dotlambda) [inline pkgs/development/python-modules/sanic/default.nix:60]:
```suggestion
```

**Comment by mweinelt** [comment]
Anyone a clue what's up with that eval error? https://gist.github.com/GrahamcOfBorg/e15c22e78501630f94d4c8d8b1461815

**Comment by mweinelt** [comment]
I'm inclined to get these fixes into directly into master and separate the python3 bump out and into `python-updates`.

File: doc/languages-frameworks/python.section.md [hunk 4]
```diff
@@ -70,19 +70,19 @@ temporary shell session with a Python and a *precise* list of packages (plus
 their runtime dependencies), with no other Python packages in the Python
 interpreter's scope.
 
-To create a Python 3.8 session with `numpy` and `toolz` available, run:
+To create a Python 3.9 session with `numpy` and `toolz` available, run:
 
 ```sh
-$ nix-shell -p 'python38.withPackages(ps: with ps; [ numpy toolz ])'
+$ nix-shell -p 'python39.withPackages(ps: with ps; [ numpy toolz ])'
```
Comment (SuperSandro2000) [inline doc/languages-frameworks/python.section.md:76]: How about we save ourselves from updating this?

```suggestion
$ nix-shell -p 'python3.withPackages(ps: with ps; [ numpy toolz ])'
```

Comment (FRidh) [reply doc/languages-frameworks/python.section.md:76]: because that does not correspond to the description of the example

**Comment by FRidh** [comment]
Is the bootstrap or minimal Python 3 updated as well?

File: doc/languages-frameworks/python.section.md (hunk 4 above)
Comment (SuperSandro2000) [reply doc/languages-frameworks/python.section.md:76]: OFC that would also be changed to only mention python 3. IMO we don't need to update half the documentation when updating the default python version.

**Comment by mweinelt** [comment]
Thanks for the pointer, it is now.

**Comment by mweinelt** [comment]
@ofborg eval

**Pull request closed without merging**